Gazebo support (draft)

mikaelarguedas opened the pull request:
**DO NOT MERGE** It was initially marked as draft but apparently something went wrong on page refresh

mikaelarguedas committed:
first step to gazebo support

rename links and joints to have semantic names
add inertia to all links
add laser and diffdrive plugins

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add scripts taken and modifies from turtlebot3_simulation

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add simulation launchfile

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
move urdf_spawner back in confbot_simulation

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas commented on confbot_description/urdf/confbot.urdf:
```diff
@@ -11,19 +11,45 @@
       <geometry>
-        <mesh filename="package://confbot_description/meshes/confbot.dae"/>
+        <!--mesh filename="package://confbot_description/meshes/confbot.dae"/-->
+        <mesh filename="file:///media/mikatchou/ROS/ros2/roscon2019/roscon2019_ws/install/share/confbot_description/meshes/confbot.dae"/>
```
This should be changed. Not sure why the resource cannot be retrieved when using `package://`

mikaelarguedas replied:
By tweaking `GAZEBO_MODEL_PATH` and accepting some additionnal error message in terinal we should be able to work around this. removed in https://github.com/Karsten1987/roscon2018/pull/19/commits/e00874ac34adb066a2e3af9021058b84201d0bf6

mikaelarguedas committed:
restore package:// path

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add dependencies not part of the nightly image

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
blacklist dependencies built from source from the test invocation

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
fix test failures

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
almost empty world

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
rebasing

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas commented on confbot_description/urdf/confbot.urdf:
```diff
@@ -32,33 +57,184 @@
+
+<gazebo reference="laser_frame">
+    <!--sensor type="gpu_ray" name="lidar"-->
+    <sensor type="ray" name="lidar">
```
I'm using ray here because `gpu_ray` returns infinity for all points even the one hitting obstacles.
also it prints this warning:
```
[gazebo-1] [Wrn] [msgs.cc:1852] Conversion of sensor type[gpu_ray] not supported.
```

chapulina replied:
That warning can be ignored.

Does your model have visuals besides collisions? The `gpu_ray` sees visuals, while the `ray` sees collisions.

Can you reproduce the issue with this demo world? https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/worlds/gazebo_ros_ray_sensor_demo.world

mikaelarguedas replied:
I'm using only primitive shapes that I dropped in the world from the Gazebo UI, I think they have both visual and collision.

Thanks for the pointer, we'll try it on the demo world to see if it's the same.

Note: I'm compiling the gazebo plugins from source without any specific flags, maybe GPU support is not enabled?

chapulina replied:
>  I think they have both visual and collision.

Yup, they do

> I'm compiling the gazebo plugins from source without any specific flags, maybe GPU support is not enabled?

There are no flags to disable GPU support, so compilation should be working correctly. At runtime, if `gzclient` works, then I'd expect the rendering sensors to work as well. You could double-check by dropping a camera in the world and seeing if the images show the shapes.

mikaelarguedas replied:
@chapulina Sorry for the delay, I fInally got a chance to try the demo world (using the dashing debs), and I'm facing the same issue:

command ran:
```
gazebo --verbose /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_ray_sensor_demo.world```
```

<details><summary>Regular ray sensor works (some non inf values)</summary>

```
$ ros2 topic echo /ray/laserscan
header:
  stamp:
    sec: 13
    nanosec: 25000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0448575019836426, 2.0313127040863037, 2.0236432552337646, 2.018503189086914, 2.0150201320648193, 2.0128238201141357, 2.0117366313934326, 2.0116827487945557, 2.0126583576202393, 2.0147311687469482, 2.0180652141571045, 2.0230026245117188, 2.030329465866089, 2.0427944660186768, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.455374836921692, 1.45494544506073, 1.4545341730117798, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 100.0, 100.0, '...']



```

</details>

<details><summary>gpu_ray sensor returns only infinity</summary>

```
$ ros2 topic echo /gpu_ray/laserscan
header:
  stamp:
    sec: 7
    nanosec: 0
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [-.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, '...']
intensities: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, '...']

```

</details>


I tried the camera demo and can see the objects successfully:
```
gazebo --verbose  /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_camera_demo.world
```

chapulina replied:
Interesting, I get very similar readings from both of them, I'm on a source install.

```
$ ros2 topic echo /gpu_ray/laserscan
header:
  stamp:
    sec: 139
    nanosec: 2000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0503170490264893, 2.033238172531128, 2.024780511856079, 2.0192136764526367, 2.0154314041137695, 2.012951612472534, 2.011847734451294, 2.0118982791900635, 2.0129361152648926, 2.0150675773620605, 2.0186455249786377, 2.0241785049438477, 2.032437562942505, 2.0490212440490723, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.454956293106079, 1.4545066356658936, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, '...']
---
```

```
$ ros2 topic echo /ray/laserscan
header:
  stamp:
    sec: 153
    nanosec: 18000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0448575019836426, 2.0313127040863037, 2.0236432552337646, 2.018503189086914, 2.0150201320648193, 2.0128238201141357, 2.0117366313934326, 2.0116827487945557, 2.0126583576202393, 2.0147311687469482, 2.0180652141571045, 2.0230026245117188, 2.030329465866089, 2.0427944660186768, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.455374836921692, 1.45494544506073, 1.4545341730117798, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 100.0, 100.0, '...']
---
```

Do the GPU readings look right on Gazebo's topic viewer (`Ctrl + T`, and you need to zoom in):

![topicviewer](https://user-images.githubusercontent.com/5751272/58994650-1cdbd280-87a6-11e9-9de3-8c731461599c.png)

I can only think it's an issue with your graphics, but I'm not sure why the RGB camera works. Are the point cloud messages working?

mikaelarguedas replied:
> I can only think it's an issue with your graphics, but I'm not sure why the RGB camera works. 

Yeah I had some issues of RViz being very slow as well. I ended up reinstalling all the graphics drivers and it seems to have fixed the issue !

Thanks for taking the time to reproduce :bowing_man: !

mikaelarguedas commented:
How I'm testing this:

ros2.repos + additionnal_repos.repos in an underlay
this repo in an overlay built with: 
```
colcon build --merge-install --cmake-args -DSECURITY=ON -DPOLICY_FILE=<PATH_TO_WORKSPACE>/src/roscon2018/confbot_security/safe_zone_publisher_policies.xml
```

Running:
In all shells start with:
`source <PATH_TO_WORKSPACE>/install/setup.bash`

Shell 1:
```
export GAZEBO_MODEL_PATH=<PATH_TO_WORKSPACE>/install/share/
ros2 launch confbot_simulation empty_world.launch.py
```

Shell 2
```
ros2 run confbot_simulation urdf_spawner.py <PATH_TO_WORKSPACE>/src/roscon2018/confbot_description/urdf/confbot.urdf
ros2 topic pub /cmd_vel geometry_msgs/Twist "{angular: {z: 1}}" -1
```

Shell 3:
```
ros2 run rviz2 rviz2 -d `ros2 pkg prefix confbot_bringup --share`/config/confbot.rviz
```

mikaelarguedas commented:
CI currently blocked by https://github.com/mikaelarguedas/ros2ci/pull/11

mikaelarguedas commented:
Things that don't work as originally expected:

GPU ray:
https://github.com/Karsten1987/roscon2018/pull/19#discussion_r271888237

Simulation time:
- we use the `gazebo_ros_factory` library to be able to spawn models in gazebo. But that libary doesnt publish the simulated time resulting in using system time everywhere anyway. `gazebo_ros_init` seems to publish the sim time, but doesnt allow to spawn model via services. Is there a way to get both the spawning interfaces and the simulation time published ?

Published transforms
- tf from `odom` to `base_link` is published by the diff_drive plugin. Regardless of the `update_rate` specified for that plugin, RViz can't display more that a laser scan every second or so due to missing transform.
```
Lookup would require extrapolation into the future.  Requested time 117.20500 but the latest data is at time 117.18300, when looking up transform from frame [base_link] to frame [odom]
```
May be related to the issue above

@Karsten1987 as you said you had a different gazebo demo available, any clue as of what should be changed here? (maybe @chapulina has some insights as well?)

mikaelarguedas committed:
dont set use_sim_time as currently doesn't impact time used

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
Revert "dont set use_sim_time as currently doesn't impact time used"

This reverts commit b79ca27fe72c74bf2c06e012c26e488e419ecee2.

mikaelarguedas committed:
use libgazebo_ros_init to have sim time published

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

chapulina commented:
> RViz can't display more that a laser scan every second or so due to missing transform

Maybe it's due to https://github.com/ros2/rviz/issues/332 ? I've seen something similar with wheel transforms on the diff drive demo, see https://github.com/ros-simulation/gazebo_ros_pkgs/pull/806#issuecomment-417496670

chapulina commented:
>  a different gazebo demo available

Maybe it was https://github.com/chapulina/dolly/ ? I've visualized the laser scans on RViz, but since the laser update rate itself is low on purpose, I probably never noticed that problem. Turn the bug into a feature :wink:

mikaelarguedas commented:
> Maybe it's due to ros2/rviz#332 ? I've seen something similar with wheel transforms on the diff drive demo, see ros-simulation/gazebo_ros_pkgs#806 (comment)

Thanks! It looks like the same issue, I tried the referenced PR enabling `message_filters` but ended up with RViz segfaulting.
@Karsten1987 I'll be away from a computer for the next couple weeks, would you mind giving this a try when you get a chance ?

Karsten1987 commented:
I've rebased this branch on top of the latest master to be able to compile to latest ros2 master.

Karsten1987 commented:
So I gave this a shot and had only very poor results. I can't get around the TF error independent on where to set `use_sim_time`. 
The robot model spawns in gazebo (without the base mesh though) and moves correctly when publishing a velocity. 
Things go very south when trying to visualize all of this in RViz. The laser scans are not shown at all and the TF error let's my mac run on 100% CPU power. So I am not sure what to do here next.
I've tried setting the the simulation time on every node, but that also doesn't yield any success.

I'll try to catch @chapulina in the next few days to sort this out offline.

Karsten1987 committed:
fix conversion error from string to bool

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

Karsten1987 committed:
actually using the sim time parameter correctly

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

Karsten1987 commented:
For setting the `use_sim_time` correctly the `LaunchConfigurations` have to be conditioned. See https://github.com/ros2/launch_ros/issues/30 for a full issue description

Karsten1987 committed:
interpret launch configuration as yaml

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

Karsten1987 commented:
we could boil it down far enough to make it work flawless when setting the fixed frame in RViz to the `laser_frame`. We've seen a constant time delay of approx 100ms which makes the TF transform fail (with sim_time enabled).
So I am not sure if it's the laser or the TF message filters which are not working.

Karsten1987 committed:
visualize the laser scan in gazebo

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

Karsten1987 committed:
use sim time

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

mikaelarguedas committed:
make laser_frame point forward

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
gazebo_ros now declares/set use_sim_time to true by default

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add camera

Need to create an optical frame to match REP103 as camera plugin uses x forward and not z
forward
Need to source Gazebo setup file manually for the camera plugin to be
found and loaded successfully
No namespace provided to plugin as it appends the sensor name to the
namespace already, otherwise results in
/confbot_camera/confobt_camera/image_raw

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
move back camera_frame on the robot base

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add rviz config for gazebo

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
add preliminary Gazebo section

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

mikaelarguedas committed:
fix config file name in instructions

Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

Karsten1987 commented on tutorial/Linux_Tutorial.md:
```diff
@@ -386,3 +386,34 @@ $ python3 safe_zone_publisher.py
+
+##### Simulating the robot in Gazebo
+
+Make sure to kill all the running nodes.
```
great opening :)

mikaelarguedas replied:
![sup](https://user-images.githubusercontent.com/6644208/59035834-6abc0d80-886e-11e9-93fa-484228365454.gif)

mikaelarguedas replied:
nothing like a  grand entrance for a tutorial :)

mikaelarguedas commented:
I'll merge this as an initial version and ticket pending issues / open follow-up PRs
